File: Pertemuan 3_4/tebak_angka.py
import random
import os

def load_best():
    if os.path.exists("best_hard.txt"):
        with open("best_hard.txt", "r") as f:
            return int(f.read())
    return None

def save_best(score):
    with open("best_hard.txt", "w") as f:
        f.write(str(score))

def play():
    min_v, max_v = 1, 1000
    secret = random.randint(min_v, max_v)
    attempts = 0

    best = load_best()

    print(f"Tebak angka antara {min_v}-{max_v}")

    while True:
        try:
            guess = int(input("Masukkan tebakan: "))
        except ValueError:
            print("Input harus angka!")
            continue

        attempts += 1

        if guess < secret:
            print("Terlalu kecil")
        elif guess > secret:
            print("Terlalu besar")
        else:
            print(f"Benar! Percobaan: {attempts}")

            if best is None or attempts < best:
                print("🎉 Best score baru!")
                save_best(attempts)
            else:
                print(f"Best score: {best}")
            break

if __name__ == "__main__":
    play()

File: Pertemuan 3_4/tebak_angka.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int load_best() {
    FILE *f = fopen("best_hard.txt", "r");
    int score;
    if (f == NULL) return -1;

    fscanf(f, "%d", &score);
    fclose(f);
    return score;
}

void save_best(int score) {
    FILE *f = fopen("best_hard.txt", "w");
    fprintf(f, "%d", score);
    fclose(f);
}

int main() {
    int min = 1, max = 1000;
    int guess, attempts = 0;

    srand(time(NULL));
    int secret = rand() % (max - min + 1) + min;

    int best = load_best();

    printf("Tebak angka antara %d-%d\n", min, max);

    while (1) {
        printf("Masukkan tebakan: ");
        scanf("%d", &guess);

        attempts++;

        if (guess < secret) {
            printf("Terlalu kecil\n");
        } else if (guess > secret) {
            printf("Terlalu besar\n");
        } else {
            printf("Benar! Percobaan: %d\n", attempts);

            if (best == -1 || attempts < best) {
                printf("Best score baru!\n");
                save_best(attempts);
            } else {
                printf("Best score: %d\n", best);
            }
            break;
        }
    }

    return 0;
}